components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -91,11 +92,15 @@ void RewardsNotificationServiceImpl::DeleteNotification(
     rewards_notification = rewards_notifications_[id];
     rewards_notifications_.erase(id);
   }
+  StoreRewardsNotifications();
+
   OnNotificationDeleted(rewards_notification);
 }
 
 void RewardsNotificationServiceImpl::DeleteAllNotifications() {
   rewards_notifications_.clear();
+  rewards_notifications_displayed_.clear();
```

Comment:
we can't clear what was displayed as then we lose functionality of show only once

Reply:
It doesn't make sense to me at all to call a function DeleteAllNotifications, but keep some of them. Anyways we use that on in ResetTheWholeState functionality on Android. We will do ifdef there and it has to be fixed in a follow up issue to reset state of ledger.

Reply:
`DeleteAllNotifications` does exactly what it should. It deletes notification. It doesn't delete notification and logs which notifications were displayed.

Reply:
temp excluded with pre-processor for Android only.
`Reset state` refactoring issue: 
https://github.com/brave/brave-browser/issues/6239